relay-server/src/services/project.rs

```diff
@@ -1309,18 +1287,6 @@ mod tests {
         }
     }
 
-    #[test]
-    fn no_error_on_disabled_project() {
-        let project_key = ProjectKey::parse("a94ae32be2584e0bbd7a4cbb95971fee").unwrap();
-        let mut project = Project::new(project_key, Arc::new(Config::default()));
-        let mut project_state = ProjectState::allowed();
-        project_state.disabled = true;
-        project.state = State::Cached(project_state.into());
-
-        let result = project.check_buckets_inner(Addr::custom().0, vec![]);
-        assert!(matches!(result, CheckedBuckets::ProjectDisabled(0)));
-    }
```

Comment:
I agree that the test did not _actually_ verify that no error was emitted, but reducing test coverage here increases vulnerability against regressions in future refactors. If there is an easy way to capture the logs in the test, as you suggested in the PR description, I would prefer that.

Reply:
Do we really want to test logging here? This is not part of the functional behavior of this function, it seems wrong to me that we're making assumptions about a specific log level.

Reply:
I'll look into this in a separate PR and then try to bring this test back. We'll have to introduce our own Test subscriber for this, which I think is going to be a useful utility once we port more unit tests over from the integration test suite. This one test I'm still torn about, as it asserts behavior that is not part of any contract or signature, but it's not a critical concern for me.